From 33f8845232bdc762fd94c57ecd0fb4d527e680d5 Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Mon, 16 Feb 2026 02:54:36 +0100
Subject: [PATCH 8/8] HACK: winegstreamer: Disable nvdecoder by default.

Seems to cause crashes during cutscenes in HSR, especially since 4.0: just fallback to software for now.
---
 dlls/winegstreamer/unixlib.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/dlls/winegstreamer/unixlib.c b/dlls/winegstreamer/unixlib.c
index 794fe01c52c..49ead38fecd 100644
--- a/dlls/winegstreamer/unixlib.c
+++ b/dlls/winegstreamer/unixlib.c
@@ -309,6 +309,14 @@ NTSTATUS wg_init_gstreamer(void *arg)
         return STATUS_UNSUCCESSFUL;
     }
 
+    /* HACK: Disable nvdecoder due to cutscenes crashes in HSR on NVIDIA */
+    GstElementFactory *nvh264dec = gst_element_factory_find("nvh264dec");
+    if (nvh264dec)
+    {
+        gst_plugin_feature_set_rank(GST_PLUGIN_FEATURE(nvh264dec), GST_RANK_NONE);
+        gst_registry_add_feature(gst_registry_get(), GST_PLUGIN_FEATURE(nvh264dec));
+    }
+
     if (!NtQueryInformationProcess(GetCurrentProcess(),
             ProcessAffinityMask, &process_mask, sizeof(process_mask), NULL))
         thread_count = popcount(process_mask);
-- 
2.54.0

